core/roboconf-agent/src/main/java/net/roboconf/agent/internal/misc/AgentUtils.java

```diff
@@ -112,6 +118,39 @@ public static void copyInstanceResources( Instance instance, Map<String,byte[]>
 
 
 	/**
+	 * Executes a script resource on a given instance.
+	 * @param instance an instance
+	 * @throws InterruptedException
+	 * @throws IOException
+	 */
+	public static List<ExecutionResult> executeScriptResources( Instance instance, String applicationName, String scopedInstanceName ) throws IOException, InterruptedException {
+
+		File dir = InstanceHelpers.findInstanceDirectoryOnAgent( instance );
+		Utils.createDirectory(dir);
+		List<File> scriptFiles = Utils.listAllFiles( dir, Constants.FILE_EXT_SCRIPT);
+		List<ExecutionResult> results = new ArrayList<ExecutionResult> ();
+		Logger logger = Logger.getLogger( AgentUtils.class.getName());
+
+		for( File script : scriptFiles) {
+			StringBuilder normalOutput = new StringBuilder();
+			StringBuilder errorOutput = new StringBuilder();
+			ExecutionResult result = new ExecutionResult(normalOutput.toString(),errorOutput.toString(), -1);
```

Comment:
It would be better to log something directly here and only when there is a script.